sharding/interface.go

```diff
@@ -24,9 +24,11 @@ type Validator interface {
 // NodesCoordinator defines the behaviour of a struct able to do validator group selection
 type NodesCoordinator interface {
 	PublicKeysSelector
-	SetNodesPerShards(eligible map[uint32][]Validator, waiting map[uint32][]Validator, epoch uint32) error
+	SetNodesPerShards(eligible map[uint32][]Validator, waiting map[uint32][]Validator, epoch uint32, updatePeers bool) error
```

Comment:
do we really need two interfaces ? can't we use the other interface for this purpose in a loop?